staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/defaults.go

```diff
@@ -66,7 +66,17 @@ func SetDefaults_CustomResourceDefinitionSpec(obj *CustomResourceDefinitionSpec)
 	if len(obj.Version) == 0 && len(obj.Versions) != 0 {
 		obj.Version = obj.Versions[0].Name
 	}
-	if len(obj.AdditionalPrinterColumns) == 0 {
+	// If the CRD uses per-version columns, apply default on per-version columns. Let the
+	// validation logic to enforce mutual exclusivity later.
+	if hasPerVersionColumns(obj.Versions) {
+		for i := range obj.Versions {
+			if len(obj.Versions[i].AdditionalPrinterColumns) == 0 {
+				obj.Versions[i].AdditionalPrinterColumns = []CustomResourceColumnDefinition{
+					{Name: "Age", Type: "date", Description: swaggerMetadataDescriptions["creationTimestamp"], JSONPath: ".metadata.creationTimestamp"},
+				}
+			}
+		}
+	} else if len(obj.AdditionalPrinterColumns) == 0 {
```

Comment:
@mbohlool had a good point on this defaulting code:

- For existing CRDs, we (the existing code) have defaulted the top-level columns to be non-nil. 
- For new CRDs (after upgrade), the top-level columns will be defaulted (new code in this PR) if a user doesn't specify any top-level/per-version columns.

In both cases, if a user decides to patch per-version columns after upgrade (with gate enabled), the PATCH request will fail the mutual-exclusive validation-- the user has to do an UPDATE.

We basically cannot avoid the first case. For the second case, @mbohlool suggested that we can do logical defaulting to improve the user experience.

I feel it's unfair to ask users to send UPDATE requests in the second case. I'm going to add the logical defaulting code.  For the first case, we have to add documentation on that.